### PR TITLE
[ValueTracking] Handle and/or of conditions in `computeKnownFPClassFromContext`

### PR DESCRIPTION
**Pull request by dtcxzyw**
Fix a typo introduced by  https://github.com/llvm/llvm-project/pull/83161.
This patch also supports decomposition of and/or expressions in `computeKnownFPClassFromContext`.
Compile-time improvement: http://llvm-compile-time-tracker.com/compare.php?from=688bb432c4b618de69a1d0e7807077a22f15762a&to=07493fc354b686f0aca79d6f817091a757bd7cd5&stat=instructions:u

### CODE REVIEW

File: llvm/lib/Analysis/ValueTracking.cpp [hunk 1]
```diff
@@ -4885,6 +4885,13 @@ static void computeKnownFPClassFromCond(const Value *V, Value *Cond,
                                         bool CondIsTrue,
                                         const Instruction *CxtI,
                                         KnownFPClass &KnownFromContext) {
+  Value *A, *B;
+  if (CondIsTrue ? match(Cond, m_LogicalAnd(m_Value(A), m_Value(B)))
+                 : match(Cond, m_LogicalOr(m_Value(A), m_Value(B)))) {
+    computeKnownFPClassFromCond(V, A, CondIsTrue, CxtI, KnownFromContext);
+    computeKnownFPClassFromCond(V, B, CondIsTrue, CxtI, KnownFromContext);
+    return;
+  }
```
Comment (nikic) [inline llvm/lib/Analysis/ValueTracking.cpp:4894]: Should we increase Depth here like in computeKnownBitsFromCond?

**Review by nikic — APPROVED** [review]
LGTM

**Comment by asmok-g** [comment]
Heads-up: [this function](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/tensor_util.cc#L292) changes behavior when optimized after this commit. Precisely when optimized + using msan on k8.

**Comment by asmok-g** [comment]
@dtcxzyw please find the attached repro https://godbolt.org/z/oPdnz87va

**Comment by dtcxzyw** [comment]
Reduced reproducer: https://godbolt.org/z/7Md6ov4oG

**Pull request merged**